genie-web/src/main/java/com/netflix/genie/web/scripts/ClusterLoadBalancerScript.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.scripts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.genie.common.dto.JobRequest;
+import com.netflix.genie.common.internal.dto.v4.Cluster;
+import com.netflix.genie.common.internal.exceptions.unchecked.GenieClusterNotFoundException;
+import com.netflix.genie.web.apis.rest.v3.controllers.DtoConverters;
+import com.netflix.genie.web.exceptions.checked.ScriptExecutionException;
+import com.netflix.genie.web.exceptions.checked.ScriptNotConfiguredException;
+import com.netflix.genie.web.properties.ClusterLoadBalancerScriptProperties;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link ManagedScript} that delegates selection of a job's cluster when more than one choice is
+ * available. See also: {@link com.netflix.genie.web.services.ClusterLoadBalancer}.
+ * <p>
+ * The contract between the script and the Java code is that the script will be supplied global variables
+ * {@code clusters} and {@code jobRequest} which will be JSON strings representing the list (array) of clusters
+ * matching the cluster criteria tags and the job request that kicked off this evaluation. The code expects the script
+ * to either return the id of the cluster if one is selected or null if none was selected.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public class ClusterLoadBalancerScript extends ManagedScript {
```

Comment:
Since you're making this a new class anyway I'd planned to rename this class / feature `ClusterSelector` instead of `ClusterLoadBalancer` as part of some of the resource data model changes we've been discussing. Might be able to kill two birds with one stone here

Reply:
So there's `ClusterLoadBalancer` interface with its implementations, and there's `ClusterLoadBalancerScript` (this newly added class that wraps a script resource).

Which one you want to rename? @tgianos

Reply:
I'd like to rename `ClusterLoadBalancer` to `ClusterSelector` as it's clearer the purpose and then as fallout of that we probably should rename the implementations as well.

Reply:
Ok will do (as last commit of this PR, to avoid making other commits hard to read)